src/App.js

```diff
@@ -14,11 +14,85 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+export class App extends React.Component {
+  state = {
+    nameOfProduct: '',
+    arrayOfProducts: [],
+  }
 
+  handleSelection = (product) => {
+    this.setState(() => ({
+      nameOfProduct: product,
+      arrayOfProducts: [],
+    }));
+  }
+
+  handleMultipleSelection = (product) => {
+    this.setState((prevState) => {
+      let newArrayOfProducts = prevState.arrayOfProducts
+        .filter(good => good !== product);
+
+      newArrayOfProducts = [...newArrayOfProducts, product];
```

Comment:
at first you remove product from the list, and then add it again to the end? There should be if...else statement